src/main/java/com/github/distributedtracing/ContinuationLocal.java

```diff
@@ -0,0 +1,170 @@
+package com.github.distributedtracing;
+
+public final class ContinuationLocal<T> {
+    private static InheritableThreadLocal<Object[]> localCtx =
+            new InheritableThreadLocal<Object[]>();
+    private static volatile int size = 0;
+
+    /**
+     * Return a snapshot of the current ContinuationLocal state.
+     */
+    public static Object[] save() {
```

Comment:
Again, also a trade-off here; Twitter solved this by using a type alias:
https://github.com/twitter/util/commit/288c79067c611be5774644cefd17d2c7ddcf28cf
I'm not sure if their construct is 100 % safe either. But the idea was to reduce allocations by not using a wrapper.
I do agree with you that the safety an encapsulating class will provide justifies the allocation overhead. Any other people with opinions?